featuretools/demo/flight.py

```diff
@@ -1,68 +1,219 @@
+    clean_data = _fill_labels(clean_data)
+
+    # Nulls for scheduled values are too problematic. Remove them.
+    clean_data = clean_data.dropna(
+        axis='rows', subset=['scheduled_dep_time', 'scheduled_arr_time'])
+
+    # Make a flight id. Define a flight as a combination of:
+    # 1. carrier 2. flight number 3. origin airport 4. dest airport
+    clean_data.loc[:, 'flight_id'] = clean_data['carrier'] + '-' + \
+        clean_data['flight_num'].apply(lambda x: str(x)) + ':' + clean_data['origin'] + '->' + clean_data['dest']
+
+    return clean_data
+
+
+def _fill_labels(clean_data):
+    labely_columns = ['arr_delay', 'dep_delay', 'carrier_delay', 'weather_delay',
+                      'national_airspace_delay', 'security_delay',
+                      'late_aircraft_delay', 'cancelled', 'diverted',
+                      'taxi_in', 'taxi_out', 'air_time']
+    for col in labely_columns:
+        clean_data.loc[:, col] = clean_data[col].fillna(0)
+
+    return clean_data
+
+
+def _reconstruct_times(clean_data):
+    """ Reconstruct departure_time, scheduled_dep_time,
+        arrival_time and scheduled_arr_time by adding known delays
+        to known times. We do:
+            - dep_time is scheduled_dep + dep_delay
+            - arr_time is dep_time + taxiing and air_time
+            - scheduled arrival is scheduled_dep + scheduled_elapsed
+    """
+    clean_data.loc[:, 'dep_time'] = clean_data['scheduled_dep_time'] + \
+        pd.to_timedelta(clean_data['dep_delay'], unit='m')
+
+    clean_data.loc[:, 'arr_time'] = clean_data['dep_time'] + \
+        pd.to_timedelta(clean_data['taxi_out'] +
+                        clean_data['air_time'] +
+                        clean_data['taxi_in'], unit='m')
+
+    clean_data.loc[:, 'scheduled_arr_time'] = clean_data['scheduled_dep_time'] + \
+        clean_data['scheduled_elapsed_time']
+    return clean_data
+
+
+def filter_data(clean_data,
+                month_filter=None,
+                categorical_filter=None):
+    if month_filter is not None:
+        tmp = False
+        for month in month_filter:
+            tmp = tmp | (clean_data['scheduled_dep_time'].apply(lambda x: x.month) == month)
+        clean_data = clean_data[tmp]
+
+    if categorical_filter is not None:
+        tmp = False
+        for key, values in categorical_filter.items():
+            tmp = tmp | clean_data[key].isin(values)
+        clean_data = clean_data[tmp]
+
+    return clean_data
+
+
+def convert(name):
+    # Rename columns to underscore
+    # Code via SO https://stackoverflow.com/questions/1175208/elegant-python-function-to-convert-camelcase-to-snake-case
+    s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', name)
+    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s1).lower()
+
+
+def make_flight_pathname(demo=True):
+    if demo:
+        filename = 'flight_dataset_sample.csv.zip'
+        key = 'bots_flight_data_2017/data_2017_jan_feb.csv.zip'
+        rows = 860457.0
```

Comment:
does this have to be a float?

Reply:
We do `rows/ 99` earlier. Not sure if the cleanest solution is to do this, `99.` or `float(rows)` when we divide.

Reply:
not sure it matters because you take the `ceil` of it, but move any casting as close as possible to where it is required